mesonbuild/interpreter.py

```diff
@@ -2307,7 +2312,10 @@ def func_configure_file(self, node, args, kwargs):
             # We use absolute paths for input and output here because the cwd
             # that the command is run from is 'unspecified', so it could change.
             # Currently it's builddir/subdir for in_builddir else srcdir/subdir.
-            values = mesonlib.get_filenames_templates_dict([ifile_abs], [ofile_abs])
+            if ifile_abs:
+                values = mesonlib.get_filenames_templates_dict([ifile_abs], [ofile_abs])
+            else:
+                values = mesonlib.get_filenames_templates_dict(None, [ofile_abs])
```

Comment:
I was actually thinking that it makes sense to allow people to pass any number of files as input, so `ifile_abs` would then come from `meson.stringlistify(kwargs.get('input', []))`, would either be N files list or an empty list, and we won't need an if condition here.